### PR TITLE
for 06/29 CRITICAL MESSAGE: EVS code update

### PR DESCRIPTION
**Pull request by Ho-ChunHuang-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by ShelleyMelchior-NOAA** [comment]
@Ho-ChunHuang-NOAA , I have picked up this PR for you. Please go ahead and merge auth EVS develop into this feature branch to pick up the changes to `parm/metplus_config/machine.conf`. If you would like, in response to the changes to `parm/metplus_config/machine.conf`, please also update the aqm driver scripts and include the changes in this PR.
`jevs_aqm_prep.sh`
`jevs_aqm_stats.sh`
`jevs_aqm_plots.sh`

**Comment by Ho-ChunHuang-NOAA** [comment]
Shelley:

NONE of environmental variables in machine.conf were defined in
jevs_aqm_*.sh.

But in jevs_aqm_*.sh, there are lines

export config=$HOMEevs/parm/evs_config/aqm/config.evs.aqm.prod
source $config

where $config includes

export metplus_verbosity="DEBUG"
export met_verbosity="2"
export log_met_output_to_metplus="yes"

Do you mean they should comment out? in
parm/evs_config/aqm/config.evs.aqm.prod.

Ho-Chun


--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by ShelleyMelchior-NOAA** [comment]
Correct, @Ho-ChunHuang-NOAA , that is my understanding. @PerryShafran-NOAA , can you confirm?

**Comment by PerryShafran-NOAA** [comment]
Yes, comment out/remove from the config file in evs_config so that ensures that what is in machine.conf will run.  It doesn't *have* to be commented out since machine.conf is used when you call run_metplus.py, but it probably should.  

Perry

**Comment by Ho-ChunHuang-NOAA** [comment]
Sheeley and Perry:

Thanks for the clarification.

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Shelley:

The update is done.

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by ShelleyMelchior-NOAA** [comment]
I ran `jevs_aqm_prep.sh`. Please review the output.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/aqm`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_prep_00.o66977988`

I can start stats next. Should I point to an archive instead of my own prep? If so, where?

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Shelley:

make sure your COMIN are the same among steps.  COMOUT will be auto set
to $COMIN/${STEP}/${COMPONENT}.

In your example, export COMIN=/lfs/h2/emc/vpppg/
noscrub/shelley.melchior/evs/v1.0


--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Shelley:

Make sure you have three days of prep 06/29, 06/20, 07/01
before running stat for 07/01.

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by ShelleyMelchior-NOAA** [comment]
Thanks, @Ho-ChunHuang-NOAA  -- so essentially you're saying I should run stats using the prep that I just generated?

**Comment by ShelleyMelchior-NOAA** [comment]
Yeah, I don't have that. So can I point to a different database of prep data for the stats run?

**Comment by Ho-ChunHuang-NOAA** [comment]
I think the default is PDY-3 = 20230630 you can add

export VDATE=20230628 & 20230630

before export maillist=

in jevs_aqm_prep.sh

Then repeat prep to have the prep for 06/28-30.  Then you can run stats
(default to 20230630).

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by ShelleyMelchior-NOAA** [comment]
I have changed COMIN to point to emc.vpppg prep. I have COMOUT writing to my $USER space. I submitted the job. I will let you know when it is complete. If this approach fails, I will attempt your suggestion. I have never gone about it that way before. Thanks for the tip.

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Shelley:

If you have any issue running stats, try export COMINaqmproc  in your
jevs_aqm_stats.sh

export COMINaqmproc=/lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/prep/aqm

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Shelley:

I added some comments for running the pull-request test in jevs_aqm_*.sh.


Hi, Perry:
I have not seen your pull-request for plots.  But please
check JEVS_AQM_PLOTS to have

export COMINaqm=${COMINaqm:-$(compath.py
${envir}/com/${NET}/${evs_ver}/stats/${COMPONENT}*/${MODELNAME}*)}

It is because you need the directory header(?) of stats output in
~/scripts/aqm/plots/exevs_aqm_grid2obs_plots.sh

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by PerryShafran-NOAA** [comment]
Hi, Ho-Chun,

For now, the plots are found here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0.  I'm not sure why they're not under the plots directory, but for now, you can point your scripts here for the updated AQM plots.  I'm running a para to the regular emc.vpppg parallel, and hope to be updating to the emc.vpppg parallel very soon, but the plots are there.  The aqm plots PR was merged some time ago, and new code winds up in the perry.shafran para.  When I change back to emc.vpppg, I'll let you know.

Perry

**Comment by ShelleyMelchior-NOAA** [comment]
Hi @Ho-ChunHuang-NOAA ! I finally have my stats processing running correctly and stats files are getting generated. Please review the output stats files and the log files.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/aqm`
log files: `/lfs/h2/emc/vpppg/noscrub/shelley.melchior/jevs_aqm_stats_00.o6*` with time stamp 20:22 and later
and `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/run_aqm_stats.o67144705`

As of 4:40p, the process is still ongoing, but you can take a look now, or wait until it is done.

**Comment by Ho-ChunHuang-NOAA** [comment]
Yes, it is at 10Z now.  Will wait until the 23Z is done.  Then I will
compare the daily stats to mine.

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Shelley:

The stats generated by you are the same as
mine, /lfs/h2/emc/physics/noscrub/ho-chun.huang/evs/v1.0/stats/aqm/aqm.20230702.

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by ShelleyMelchior-NOAA** [comment]
Fantastic -- plots next.

**Comment by ShelleyMelchior-NOAA** [comment]
@Ho-ChunHuang-NOAA  -- plots processing is complete.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/aqm/atmos.20230702`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_plots_00.o67154009`

If this looks good, please go ahead and do a sync fork w/ `feature/drivermodule` as Perry merged the subseasonal PR today. Then I can merge your PR, too!

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Shelley:

Two things.

First, for this PR that is to create and use the
~/modulefiles/aqm/aqm_*.sh.  The graphics have been created (as those in
/lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/atmos.20230702) without a
problem.  Thus, I believe the PR can be merage and close.

Second, I need Perry's comment on the types of graphics created.
I only saw the *ozmax* and *pmave* graphics created for the current
authoritative repository EVS for aqm plots step.  I did not see the hourly
*ozone* *pm25* graphics.  Something is still not right.

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by PerryShafran-NOAA** [comment]
Oh dear, I am not sure what happened.  When I submitted all the changes for AQM plots, I was had all the plots for hourly ozone and PM2.5.  But now my current runs don't have them either.  I'm not sure why so I'll definitely be checking into that.  It probably has to do with missing stat files early in the period.  

However, that has nothing to do with this particular PR, so I feel that it can be merged at this time.

Perry

**Comment by AliciaBentley-NOAA** [comment]
Hi Perry and Ho-Chun,

I believe that there are currently two EVS parallels running
simultaneously. Any chance that these AQM scripts were accidentally put in
the other EVS parallel, and that's why we can't find them now? I think this
issue will be fixed soon when everything is run using the emc.vpppg
parallel again, but I figured that I would ask in the meantime. I don't
know why the alternative (that someone deleted these files in a non-AQM PR)
would ever happen, but it may have.

Take care,
Alicia
___________________________________
*Alicia M. Bentley, Ph.D.*
Global Verification Project Lead
Model Evaluation Group (MEG) <https://www.emc.ncep.noaa.gov/users/meg/home/>
NOAA/NWS/NCEP/EMC (VPPPG Branch)
Work Phone: ‪(301) 683-3783‬

**Comment by PerryShafran-NOAA** [comment]
No - the new scripts are definitely running in the parallel in perry.shafran.  I know what the issue is and I've seen it, it has to do with missing stat files.  

Perry

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, SHelley:

Can you re-ran your plots scripts by adding "export VDATE=20230702'
in jevs_aqm_plots.sh?

You created the plots around "Jul  5 22:30", but hourly stats were created
around"Jul  5 23:01".  That may be the reason we did not see the hourly
figures.

For example, /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_plots_00.o67154009,
line 836.

============================================================
Hi, Perry, a suggestion, in exevs_aqm_grid2obs_plots.sh
 35         if [ -e
${COMINaqm}.$DAY/evs.stats.${COMPONENT}_${biasc}.${RUN}.${VERIF_CASE}_${aqmtyp}.v${DAY}.stat
]
 36         then
 37         cp
${COMINaqm}.$DAY/evs.stats.${COMPONENT}_${biasc}.${RUN}.${VERIF_CASE}_${aqmtyp}.v${DAY}.stat
$STATDIR
 38         fi
 39
 40         sed "s/$model1/${aqmtyp}_${biasc}/g"
$STATDIR/evs.stats.${COMPONENT}_${biasc}.${RUN}.${VERIF_CASE}_${aqmtyp}.v${DAY}.stat
> $STATDIR/evs.stats.${aqmtyp}_${bias
 c}.${RUN}.${VERIF_CASE}.v${DAY}.stat


Maybe move the "sed" step inside the if loop to execute the replacement
only when the file exists, else print a warning message for missing files.


You still need to deal with when the plot step can be executed after the
hourly stats generated.


Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by PerryShafran-NOAA** [comment]
Ho-Chun:

That is a good idea; that all should be within the if loop.  

Perry

**Comment by ShelleyMelchior-NOAA** [comment]
It is strange that the stats timestamp is later than the plots. I definitely did not run the plots until after @Ho-ChunHuang-NOAA declared the stats looked OK.
I can re-run w/ VDATE=20230702, but should I wait for some changes from @PerryShafran-NOAA  first?

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Shelley:

I checked the stats produced in your output directory, but the stats used
in plots step are coming from

export COMINaqm=/lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/aqm/aqm

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Shelley:

Good suggestion, maybe wait after Perry said it is okay to run.  After all,
it is for the last 31 days graphics and the data of previous 30 days should
be available in  /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/aqm,

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Perry:

There is an error message at the bottom of
Cactus:/lfs/h2/emc/stmp/shelley.melchior/evs_test/prod/tmp/jevs_aqm_grid2obs_plots_00.67154009.cbqs01/plots/logs/EVS_plotting_job_20230705-220700_8013.out

for  MODELS: ['ozone_raw', 'ozone_bc']

07/05 22:07:08.081 (lead_average.py:1371) DEBUG:
========================================
07/05 22:07:08.082 (df_preprocessing.py:71) INFO: Looking for stat files in
/lfs/h2/emc/stmp/shelley.melchior/evs_test/prod/tmp/jevs_aqm_grid2obs_plots_00.67154009.cbqs01/stats
using the template:
evs.stats.{MODEL}.atmos.grid2obs.v{valid?fmt=%Y%m%d}.stat
07/05 22:07:08.968 (df_preprocessing.py:97)* ERROR:* Called from
filter_by_var_name:
07/05 22:07:08.968 (df_preprocessing.py:98) *ERROR:* Empty Dataframe.
Continuing onto next plot...
07/05 22:07:08.968 (df_preprocessing.py:99) INFO:
========================================

Can you take a look? There are data in

Cactus:/lfs/h2/emc/stmp/shelley.melchior/evs_test/prod/tmp/jevs_aqm_grid2obs_plots_00.67154009.cbqs01/stats

that can be used for this plot.

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by PerryShafran-NOAA** [comment]
I'm pretty sure this probably relates to the error that I've been talking about above.  I think you are right that if I move the "sed" inside the if block, this might cease to be an issue.  But I haven't tested that.  Top priority today is to get the parallels on track from the machine switch, so I might not have a lot of time to look at this today.  But if you want to make those changes in the script and test, that might be useful.  

Perry

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Perry:

I may be wrong but the changes of sed statements is not the critical issue
for missing hourly figures.  Even we missed one day, would that stop the
plot of hourly-last_31_days ozone and pm25?  Actually there is data from
06/04-07/02 in Shelley's runtime data directory (after sed).

It probably has something to do with the python plotting code itself.

07/05 22:07:08.968 (df_preprocessing.py:97) ERROR: Called from
filter_by_var_name:
07/05 22:07:08.968 (df_preprocessing.py:98) ERROR: Empty Dataframe.
Continuing onto next plot...

Ho-Chun
p.s. Are you getting duplicate emails if I include you in the ccing?

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by PerryShafran-NOAA** [comment]
It's not that there is no data, but if the *first* day is missing, then the whole thing dies.  That is because when it seds a file that does not exist, it creates a new file with 0 data in it.  Then, when it sees an empty file, the process is stopped and a plot is not created.  

Thus, if you put the sed inside the if block, there will be no 0 length files to look at and the plots should be made.  That is my theory though.  

Perry

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Perry:

Thanks for the clarification.  I will make the changes, then ask Shelley to
test it again.

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Shelley:

(1) I sync the fork and merge the changes to branch feature/drivermodule
(2) I modified scripts/aqm/plots/exevs_aqm_grid2obs_plots.sh to
avoid generating empty stat files for plotting.
(3) I modified jobs/aqm/plots/JEVS_AQM_PLOTS for correct setting of default
COMINaqm

My test on Dogwood for the above changes *FAILED* to produce hourly
graphics.

(1) After the updates, I did not generate empty stat file in
/lfs/h2/emc/stmp/ho-chun.huang/evs_test/prod/tmp/jevs_aqm_grid2obs_plots_00.80753965.dbqs01/stats
Regardless, some days are missing.

(2) But there are warning message (
/lfs/h2/emc/stmp/ho-chun.huang/evs_test/prod/tmp/jevs_aqm_grid2obs_plots_00.80753965.dbqs01/plots/logs/EVS_plotting_job_20230706-163249_47388.out
)

"WARNING: The stat file for obs does not exist in
/lfs/h2/emc/stmp/ho-chun.huang/evs_test/prod/tmp/jevs_aqm_grid2obs_plots_00.80753965.dbqs01/prune/SL1L2_OZCON1_Southeast_TEST/tmpd9a270b5f57d4310b2298f8ff063376c."

I found all CTC_* and SL1L2_* directory in
/lfs/h2/emc/stmp/ho-chun.huang/evs_test/prod/tmp/jevs_aqm_grid2obs_plots_00.80753965.dbqs01/
*prune* are EMPTY.  *Perry, is it normal?*

"WARNING: You might check whether the stats_dir
(/lfs/h2/emc/stmp/ho-chun.huang/evs_test/prod/tmp/jevs_aqm_grid2obs_plots_00.80753965.dbqs01/stats)
includes obs data according to the output_base template, given domain,
variable, etc"

I do not know the meaning of this message, Perry can you help?

(3)  Despite empty directories in (2), the *ozmax8* and *pmave" has been
generated.
(/lfs/h2/emc/ptmp/ho-chun.huang/evs/v1.0/plots/aqm/atmos.20230703)

Perry, I recall you mentioned an additional column of data produced in
met/v11.#.#.  Can you check the python code to handle the hourly stat that
includes the update?

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by PerryShafran-NOAA** [comment]
It would seem that the issue of reading the first day remains, even if it's 0 length.  I'm not sure why.  Shorten the length of your verification from PDYm31 to PDYm15 and see what happens.  There was a long period where I do not see aqm stats on either machine.

Perry

**Comment by PerryShafran-NOAA** [comment]
@Ho-ChunHuang-NOAA I actually found the problem.  

I didn't make corrections in the settings.py file in the ush directory for the hourly ozone and PM2.5 cases (though I did for OZMAX8 and PMAVE).  Please copy the file /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs_aqm_test/EVS/ush/aqm/settings.py, test that, and then add it to this PR. 

Perry

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Shelley:

Please update the branch feature/drivermodule.

Use current ~/dev/drivers/scripts/aqm/plots/jevs_aqm_plots.sh
(1) Update export HOMEevs=
(2) Update export COMIN=

Then submit the batch job.  The output will be
in ${COMIN}/plots/aqm/atmos.${VDATE}.

I have tested the plots step and it generated the hourly and daily
graphics after Perry's update.

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by ShelleyMelchior-NOAA** [comment]
I just re-submitted the plots job on dogwood following the code update to `feature/drivermodule` and the stats archive for the para on dogwood catching up after the machine switch.
output log file: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_plots_00.o80930318`

**Comment by ShelleyMelchior-NOAA** [comment]
The re-run is complete. Please review the png images in the tar file.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/aqm/atmos.20230704`

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Shelley:

The plot step is a success with both hourly and daily figures generated.
You can view the figures here
<https://www.emc.ncep.noaa.gov/users/verification/air_quality/aqm/evs/>.

The ozone "Forecast Mean and Observation Mean" still does not have the obs
mean.  But this is the work for a separate pull-request from Perry.

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

File: parm/evs_config/aqm/config.evs.aqm.prod [hunk 1]
```diff
@@ -37,9 +37,9 @@ export end_date=$VDATE
 #metplus_verbosity:         DEBUG, INFO, WARN, ERROR (most verbose is DEBUG)
 #met_verbosity:             MET verbosity for logging, 1-5 (most verbose is 5)
 #log_met_output_to_metplus: log MET output to METplus log, yes or no
-export metplus_verbosity="DEBUG"
-export met_verbosity="2"
-export log_met_output_to_metplus="yes"
+# export metplus_verbosity="DEBUG"
+# export met_verbosity="2"
+# export log_met_output_to_metplus="yes"
 ## DATA DIRECTIVE SETTINGS
```
Comment (ShelleyMelchior-NOAA) [inline parm/evs_config/aqm/config.evs.aqm.prod:40]: @PerryShafran-NOAA , @Ho-ChunHuang-NOAA 
Should lines 40-42 be wholesale removed, or is leaving them in, but commented out, OK?

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Shelley:

Deletion completed.  Please update the branch

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Review by ShelleyMelchior-NOAA — COMMENTED** [review]
Code review looks solid. Changes are as expected. Please do one more sync fork for `feature/drivermodule` and then I can merge the PR.

**Comment by Ho-ChunHuang-NOAA** [comment]
Sync fork done.

Ho-Chun

--

Ho-Chun Huang, Ph.D.

Physical Scientist III, Contractor with Lynker in Support of

NOAA/NWS/NCEP/EMC, U.S. Department of Commerce

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Review by ShelleyMelchior-NOAA — APPROVED** [review]

**Pull request merged**